src/util/ConfigManager/ConfigManager.cpp

```diff
@@ -30,74 +30,90 @@
 #include "util/ConfigManager/generated/ConfigShorthandParser.h"
 #include "util/Exception.h"
 #include "util/StringUtils.h"
+#include "util/TypeTraits.h"
 #include "util/antlr/ANTLRErrorHandling.h"
 #include "util/json.h"
 
 namespace ad_utility {
 // ____________________________________________________________________________
-template <typename ReturnPointer>
-std::vector<std::pair<std::string, ReturnPointer>>
+template <typename ReturnReference>
+requires std::same_as<ReturnReference, ConfigOption&> ||
+         std::same_as<ReturnReference, const ConfigOption&>
+std::vector<std::pair<std::string, ReturnReference>>
 ConfigManager::configurationOptionsImpl(auto& configurationOptions,
                                         std::string_view pathPrefix) {
-  std::vector<std::pair<std::string, ReturnPointer>> collectedOptions;
+  std::vector<std::pair<std::string, ReturnReference>> collectedOptions;
```

Comment:
> We ideally want `pair<string_view, ReturnReferene>` (no point in copying the string).

I think, there is misunderstanding here. With our current concept, of sub manager holding their configuration options and the top manager not saving pointer/references to the configuration options of sub managers, it's not possible to have a working list of all configuration option in the form of `pair<string_view, reference`.
This is, because `string_view` is a view of an already existing string. But the strings for configuration options in sub managers are created **inside** `configurationOptionsImpl`.
For example: If we have a sub manager with the json path `[objects][tea]`, which contains an option under `[brand]`, than the string of path `[objects][tea][brand]` doesn't exist.

> Most notably, we don't need the `unique_ptr`s anymore which gets rid of the nullptr checking and dereferencing at all.

That sounds very nice. I'm gonna note that down

Reply:
Thank you for clarifying this.
In this case we should leave everything as is (map from string to unique_ptr is the preferred/most efficient pattern when we only need stability for the values and not the keys, but that doesn't really matter for the `configOptions`.

Reply:
If you say so.